src/main/java/everyos.browser.webicitybrowser/everyos/browser/webicitybrowser/gui/ui/OverlyingBlockComponentUI.java

```diff
@@ -1,72 +1,103 @@
 import everyos.engine.ribbon.core.rendering.Renderer;
+import everyos.engine.ribbon.core.shape.Dimension;
+import everyos.engine.ribbon.core.shape.Location;
 import everyos.engine.ribbon.core.shape.Rectangle;
+import everyos.engine.ribbon.core.shape.SizePosGroup;
 import everyos.engine.ribbon.core.ui.ComponentUI;
+import everyos.engine.ribbon.core.ui.UIDirective;
 import everyos.engine.ribbon.core.ui.UIManager;
 import everyos.engine.ribbon.ui.simple.SimpleBlockComponentUI;
+import everyos.engine.ribbon.ui.simple.appearence.Appearence;
+import everyos.engine.ribbon.ui.simple.helper.ComputedChildrenHelper;
+import everyos.engine.ribbon.ui.simple.helper.StringWrapHelper;
+import everyos.engine.ribbon.ui.simple.layout.InlineBlockLayout;
+import everyos.engine.ribbon.ui.simple.layout.InlineLayout;
 
 public class OverlyingBlockComponentUI extends SimpleBlockComponentUI {
 
+    private Appearence appearence;
     private OverlyingBlockComponent component;
-    private Component contentPane;
-    private ComponentUI contentPaneUI;
+    private InlineBlockLayout layout;
 
     public OverlyingBlockComponentUI(Component c, ComponentUI parent) {
         super(c, parent);
-    }
-
-    //TODO: Fix merge conflicts
-    
-    /*@Override
-    protected ComponentUI[] calcChildren(UIManager uimgr) {
-        component = getComponent();
-        if (component.getChildren().length < 1) return new ComponentUI[0];
-        contentPane = component.getChildren()[0];
-        contentPane.unbindAll();
-        contentPaneUI = uimgr.get(contentPane, getParent());
-        contentPane.bind(contentPaneUI);
-        return new ComponentUI[]{contentPaneUI};
+        component = (OverlyingBlockComponent) c;
+        this.appearence = new OverlyingBlockComponentUIAppearence();
+        layout = (InlineBlockLayout) getLayout();
+        layout.setConsiderChildren(false);
     }
 
     @Override
-    protected ComponentUI[] getChildren() {
-        return new ComponentUI[]{contentPaneUI};
+    public Appearence getAppearence() {
+        return appearence;
     }
 
-    @Override
-    protected void paintUI(Renderer r) {
-        paintMouse(r);
+    private class OverlyingBlockComponentUIAppearence implements Appearence {
+
+        private Dimension bounds;
+
+        @Override
+        public void render(Renderer r, SizePosGroup sizepos, UIManager uimgr) {
+//            sizepos.move(strwidth+r.getFontPaddingHeight(), true);
+//            sizepos.setMinLineHeight(r.getFontHeight());
+
+            this.bounds = sizepos.getSize();
+            layout.renderChildren(r, sizepos, uimgr);
+
+//            contentPaneUI = uimgr.get(component.contentView, OverlyingBlockComponentUI.this);
+//            contentPaneUI.render(r, sizepos, uimgr);
+//            contentPaneUI.directive(SizeDirective.of(new Location(1, 0, 1, 0)).getDirective());
+        }
 
-        if(component.isInvisible()) return;
+        private float progress = 0;
```

Comment:
Define this before any methods (i.e. right below where we define bounds).

Reply:
Sure, I sometimes like defining variables used by only one method right above it.

It's okay though I'll move it up